src/_pytest/reports.py

```diff
@@ -368,17 +417,28 @@ def from_item_and_call(cls, item: Item, call: CallInfo[None]) -> TestReport:
                 if excinfo.value._use_item_location:
                     path, line = item.reportinfo()[:2]
                     assert line is not None
-                    longrepr = os.fspath(path), line + 1, r.message
+                    longrepr = (os.fspath(path), line + 1, r.message)
                 else:
                     longrepr = (str(r.path), r.lineno, r.message)
+            elif isinstance(excinfo.value, BaseExceptionGroup):
+                if excinfo.value.exceptions and any(
+                    isinstance(exception, skip.Exception)
+                    for exception in excinfo.value.exceptions
+                ):
```

Comment:
The condition we actually want is
```suggestion
            elif isinstance(excinfo.value, BaseExceptionGroup) and (
                excinfo.value.split(skip.Exception)[1] is None
            ):
```
to check that the group contains _only_ skip exceptions, while handling any number of levels of nested groups.  (and we should test the multiply-nested case too)

I'd further suggest merging this with the skip handler above, such that we use item location if _all_ have use-item-location else r.path.